calendar-bundle/src/Resources/contao/classes/Events.php

```diff
@@ -103,8 +103,50 @@ protected function getAllEvents($arrCalendars, $intStart, $intEnd, $blnFeatured
 			return array();
 		}
 
+		$this->collectCalendars($arrCalendars, $intStart, $intEnd, $blnFeatured);
+
+		// HOOK: modify the result set
+		if (isset($GLOBALS['TL_HOOKS']['getAllEvents']) && \is_array($GLOBALS['TL_HOOKS']['getAllEvents']))
+		{
+			foreach ($GLOBALS['TL_HOOKS']['getAllEvents'] as $callback)
+			{
+				$this->import($callback[0]);
+				$this->arrEvents = $this->{$callback[0]}->{$callback[1]}($this->arrEvents, $arrCalendars, $intStart, $intEnd, $this);
+			}
+		}
+
+		return $this->arrEvents;
+	}
+
+	/**
+	 * Collect the matching calendars.
+	 *
+	 * @param array   $arrCalendars
+	 * @param integer $intStart
+	 * @param integer $intEnd
+	 * @param boolean $blnFeatured
+	 */
+	protected function collectCalendars($arrCalendars, $intStart, $intEnd, $blnFeatured)
+	{
 		$this->arrEvents = array();
 
+		// HOOK: add custom logic
+		if (isset($GLOBALS['TL_HOOKS']['collectCalendars']) && \is_array($GLOBALS['TL_HOOKS']['collectCalendars']))
+		{
+			foreach ($GLOBALS['TL_HOOKS']['collectCalendars'] as $callback)
+			{
+				if (($this->arrEvents = System::importStatic($callback[0])->{$callback[1]}($arrCalendars, $intStart, $intEnd, $blnFeatured, $this)) === false)
+				{
+					continue;
+				}
+
+				if (\is_array($this->arrEvents))
+				{
+					return;
+				}
+			}
+		}
```

Comment:
imho it is difficult or even unnecessary to provide hooks/events for every conceivable use-case. May be in your case it's better to implement your own output modules instead?

Reply:
It is not my intention to implement a HOOK for every application. My idea is that the HOOK is executed first in the message list module and that one can implement one's own logic through it. Because the existing HOOK is simply too late at this point.